src/main/java/redis/clients/jedis/util/JedisThreadPool.java

```diff
@@ -12,19 +12,68 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import redis.clients.jedis.Protocol;
 
 /**
  * This class is used to build a thread pool for Jedis.
  */
-public class JedisThreadPoolBuilder {
-    private static final Logger log = LoggerFactory.getLogger(JedisThreadPoolBuilder.class);
+public class JedisThreadPool {
+
+    private static final Logger log = LoggerFactory.getLogger(JedisThreadPool.class);
 
     private static final RejectedExecutionHandler defaultRejectHandler = new AbortPolicy();
 
-    public static PoolBuilder pool() {
+    public static PoolBuilder poolBuilder() {
         return new PoolBuilder();
     }
 
+  /**
+   * The following are the default parameters for the multi node pipeline executor
+   * Since Redis query is usually a slower IO operation (requires more threads),
+   * so we set DEFAULT_CORE_POOL_SIZE to be the same as the core
+   */
+  private static final long DEFAULT_KEEPALIVE_TIME_MS = 60000L;
+  private static final int DEFAULT_BLOCKING_QUEUE_SIZE = Protocol.CLUSTER_HASHSLOTS;
+  private static final int DEFAULT_CORE_POOL_SIZE = Runtime.getRuntime().availableProcessors();
+  private static final int DEFAULT_MAXIMUM_POOL_SIZE = Runtime.getRuntime().availableProcessors() * 2;
+  private static ExecutorService executorService = null;
+
+  public static ExecutorService getThreadPool() {
+    return executorService;
+  }
+
+  /**
+   * Provide an interface for users to set executors themselves.
+   * @param executor the executor
+   */
+  public static void setThreadPool(ExecutorService executor) {
+    if (executorService != executor && executorService != null) {
+      executorService.shutdown();
+    }
+    executorService = executor;
+  }
+
+  /**
+   * Create thread pool with default configurations.
+   * @return thread pool
+   */
+  public static ExecutorService createDefaultThreadPool() {
+    return JedisThreadPool.poolBuilder()
+      .setCoreSize(DEFAULT_CORE_POOL_SIZE)
+      .setMaxSize(DEFAULT_MAXIMUM_POOL_SIZE)
+      .setKeepAliveMillSecs(DEFAULT_KEEPALIVE_TIME_MS)
+      .setThreadNamePrefix("jedis-multi-node-pipeline")
```

Comment:
I guess you mean to make it the default thread pool of Jedis, then we need to modify the name to `jedis-default-xxx` instead of `jedis-multi-node-pipeline`.

Before, I put it in `MultiNodePipeline.java` so that it only appears where it is used, and these thread pool parameters are prepared for the `jedis-multi-node-pipeline` scenario.